### ISSUE TITLE
Support questions with no explanation

### ISSUE THREAD

**Issue by svpino** [underfitted/bnomial-daily#59]
This should not happen, but it's better to be safe than sorry.

Let's assume there may be questions that don't return an explanation from the database (either the field is not present or its value is null.)

Let's make sure the frontend can handle that.

Do we need to revisit the messages that we display when the user submits an answer to make sure we don't promise an explanation?

**Comment by haltakov** [comment]
I tried it out and the code handles this case without a problem. The messages, however, don't make much sense then...

https://github.com/underfitted/bnomial-daily/blob/e9b17117493d5e7c2a68065bc72407b666766d2d/src/components/Quiz/Answer.tsx#L33

**Comment by svpino** [comment]
Okay, I thought about this more and what needs to happen is something slightly different.

If there's no explanation from the API, we need to create our own "canned" answer on the client side. Something simple but good enough for the person to leave knowing exactly the correct answer.

_**Note:** We will only display this canned answer if the user's performance on this question is less than 100%. If they get the question correctly, there's no need to display this canned answer. **However**, we will show the explanation (if any) regardless of how the user performs._

Here is what the process will look like:

1. User answers the question.
2. If user's performance = 100%, display the random congratulatory sentence and explanation from the API (if any.)
3. If user's performance < 100%, display the random sentence and explanation from the API or the canned answer.

### Canned Answer Example 1 (2+ choices are the correct answer)

The following choices are the correct answer for this question:

* When using ReLU, any negative inputs turn the function's gradient to zero, allowing backpropagation to converge faster and helping the network learn.
* Neural networks are easier to optimize when their behavior is as linear as possible. ReLU acts mainly as a linear function, improving our ability to optimize neural networks.

### Canned Answer Example 2 (1 choice is the correct answer)

The following choice is the correct answer for this question:

* When using ReLU, any negative inputs turn the function's gradient to zero, allowing backpropagation to converge faster and helping the network learn.

## Random sentences

We can still use the random sentences that we have for when the user misses out completely **and** there's an explanation from the API.

But if there's no explanation, we won't display these sentences, and instead, go with the canned answer directly.

Let's use the same blue color that we are using today for the messages to display the introduction to the canned answer (i.e "The following choices are the correct answer for this question")

Let me know if this makes sense.

**Comment by haltakov** [comment]
OK, it seems I didn't get your idea from the beginning, but it makes sense now. I thought that you just want to have the app not break, but the idea with the canned answers is good.

I think we should strive to always have an explanation even if it is a short one, but this may be a good alternative for some question types, where we don't need to just repeat the answers.

I'll implement this now.
